### PR TITLE
Api 8652 organization transformer continued

### PR DESCRIPTION
**Pull request by ilaflamme-va**

### CODE REVIEW

File: vista-fhir-query/src/main/java/gov/va/api/health/vistafhirquery/service/controller/organization/ExtensionFactory.java [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+
+import java.math.BigDecimal;
+import java.util.Map;
+
+import static java.util.Collections.singletonList;
+
+@AllArgsConstructor(staticName = "of")
+class ExtensionFactory {
+  LhsLighthouseRpcGatewayResponse.FilemanEntry entry;
+  Map<String, Boolean> yesNo;
```
Comment (jhulbert-va) [inline vista-fhir-query/src/main/java/gov/va/api/health/vistafhirquery/service/controller/organization/ExtensionFactory.java:18]: I don't think the factory should know about the values (e.g. yes, no). I think this should be something like `toBoolean` or something like that instead.

File: vista-fhir-query/src/main/java/gov/va/api/health/vistafhirquery/service/controller/organization/ExtensionFactory.java [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+  public Extension ofQuantity(String fieldNumber, String unit, String system) {
+    var value = entry.internal(fieldNumber);
+    if (value.isEmpty()) {
+      return null;
+    }
+    BigDecimal quantity;
+    try {
+      quantity = new BigDecimal(value.get());
+    } catch (NumberFormatException e) {
+      return null;
```
Comment (jhulbert-va) [inline vista-fhir-query/src/main/java/gov/va/api/health/vistafhirquery/service/controller/organization/ExtensionFactory.java:59]: Not sure we should be returning null on that failure, but if we are we should for sure log something too.

File: vista-fhir-query/src/main/java/gov/va/api/health/vistafhirquery/service/controller/organization/R4OrganizationTransformer.java [hunk 3]
```diff
@@ -334,6 +453,19 @@ private Address address(
             .build());
   }
 
+  private Extension payerExtension(Optional<String> value) {
+    if (value.isEmpty()) {
+      return null;
+    }
+    return Extension.builder()
```
Comment (jhulbert-va) [inline vista-fhir-query/src/main/java/gov/va/api/health/vistafhirquery/service/controller/organization/R4OrganizationTransformer.java:460]: Should reference be added to `ExtensionFactory`?

File: vista-fhir-query/src/main/java/gov/va/api/health/vistafhirquery/service/controller/organization/ExtensionFactory.java (hunk 2 above)
Comment (bschofield-va) [reply vista-fhir-query/src/main/java/gov/va/api/health/vistafhirquery/service/controller/organization/ExtensionFactory.java:59]: agreed... we're eating and masking a data problem. Explode and die.

File: vista-fhir-query/src/main/java/gov/va/api/health/vistafhirquery/service/controller/organization/ExtensionFactory.java [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+import java.util.Map;
+
+import static java.util.Collections.singletonList;
+
+@AllArgsConstructor(staticName = "of")
+class ExtensionFactory {
+  LhsLighthouseRpcGatewayResponse.FilemanEntry entry;
+  Map<String, Boolean> yesNo;
+
+  public Extension ofBoolean(String fieldNumber, String url) {
```
Comment (bschofield-va) [inline vista-fhir-query/src/main/java/gov/va/api/health/vistafhirquery/service/controller/organization/ExtensionFactory.java:20]: but ... I wouldn't name it so generically. This isn't a generic boolean, it's very much a boolean specifically for yes/no situations. Call it `ofYesNo(...)`

File: vista-fhir-query/src/main/java/gov/va/api/health/vistafhirquery/service/controller/organization/OrganizationCoordinates.java [hunk 5]
```diff
@@ -8,7 +8,7 @@ public static RecordCoordinates insuranceCompany(String ien) {
     return RecordCoordinates.builder().file("36").ien(ien).build();
   }
 
-  public static RecordCoordinates payor(String ien) {
+  public static RecordCoordinates payer(String ien) {
```
Comment (bschofield-va) [inline vista-fhir-query/src/main/java/gov/va/api/health/vistafhirquery/service/controller/organization/OrganizationCoordinates.java:11]: I've deleted this class in my PR, so look for conflicts.

File: vista-fhir-query/src/main/java/gov/va/api/health/vistafhirquery/service/controller/organization/R4OrganizationTransformer.java [hunk 6]
```diff
@@ -286,17 +292,130 @@ private Address address(
...
+            extensions.ofBoolean(
+                InsuranceCompany.ATT_REND_ID_BILL_SEC_ID_INST,
+                "http://va.gov/fhir/StructureDefinition/organization-attendingRenderingProviderSecondaryIDInstitutionalRequired"),
+            extensions.ofBoolean(
+                InsuranceCompany.PRINT_SEC_TERT_AUTO_CLAIMS_,
+                "http://va.gov/fhir/StructureDefinition/organization-printSecTertAutoClaimsLocally"),
+            extensions.ofBoolean(
+                InsuranceCompany.PRINT_SEC_MED_CLAIMS_W_O_MRA_,
+                "http://va.gov/fhir/StructureDefinition/organization-printSecMedClaimsWOMRALocally"))
+        .collect(Collectors.toList());
```
Comment (bschofield-va) [inline vista-fhir-query/src/main/java/gov/va/api/health/vistafhirquery/service/controller/organization/R4OrganizationTransformer.java:398]: filter the ^ possible nulls away

Comment (bschofield-va) [reply vista-fhir-query/src/main/java/gov/va/api/health/vistafhirquery/service/controller/organization/R4OrganizationTransformer.java:398]: ^ do this everywhere `Stream...toList()` is used

File: vista-fhir-query/src/main/java/gov/va/api/health/vistafhirquery/service/controller/organization/ExtensionFactory.java (hunk 1 above)
Comment (ilaflamme-va) [reply vista-fhir-query/src/main/java/gov/va/api/health/vistafhirquery/service/controller/organization/ExtensionFactory.java:18]: If future us needs to make this, then we'll handle it then.

**Review by jhulbert-va — APPROVED** [review]

**Pull request merged**